### ISSUE TITLE
GLTextItem.py raises exception on Python 3.10

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#1887]
Extract from : https://kojipkgs.fedoraproject.org//work/tasks/2585/71512585/build.log
```
+ xvfb-run -a py.test-3.10 -k 'not (test_reload)'
============================= test session starts ==============================
platform linux -- Python 3.10.0b3, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /builddir/build/BUILD/pyqtgraph-0.12.2
collected 92 items
examples/test_examples.py ................................s............. [ 50%]
...................s.............F............                           [100%]
=================================== FAILURES ===================================
____________________ testExamples[ GLTextItem.py - PyQt5 ] _____________________
Error while drawing item <pyqtgraph.opengl.items.GLTextItem.GLTextItem object at 0x3ff89c8f250>.
Error while drawing item <pyqtgraph.opengl.items.GLTextItem.GLTextItem object at 0x3ff89c8f370>.
/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/GLViewWidget.py:275: RuntimeWarning: 
Traceback (most recent call last):
  File "<stdin>", line 11, in <module>
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/Qt/__init__.py", line 407, in exec_
    return app.exec() if hasattr(app, 'exec') else app.exec_()
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/GLViewWidget.py", line 254, in paintGL
    self.drawItemTree(useItemNames=useItemNames)
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/GLViewWidget.py", line 294, in drawItemTree
    self.drawItemTree(i, useItemNames=useItemNames)
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/GLViewWidget.py", line 275, in drawItemTree
    debug.printExc()
  --- exception caught here ---
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/GLViewWidget.py", line 272, in drawItemTree
    i.paint()
  File "/builddir/build/BUILD/pyqtgraph-0.12.2/pyqtgraph/opengl/items/GLTextItem.py", line 79, in paint
    painter.drawText(text_pos[0], text_pos[1], self.text)
TypeError: arguments did not match any overloaded call:
  drawText(self, Union[QPointF, QPoint], str): argument 1 has unexpected type 'numpy.float64'
  drawText(self, QRectF, int, str): argument 1 has unexpected type 'numpy.float64'
  drawText(self, QRect, int, str): argument 1 has unexpected type 'numpy.float64'
  drawText(self, QRectF, str, option: QTextOption = QTextOption()): argument 1 has unexpected type 'numpy.float64'
  drawText(self, QPoint, str): argument 1 has unexpected type 'numpy.float64'
  drawText(self, int, int, int, int, int, str): argument 1 has unexpected type 'numpy.float64'
  drawText(self, int, int, str): argument 1 has unexpected type 'numpy.float64'
  debug.printExc()
Failed Text Example Test Located in GLTextItem.py
```

**Comment by j9ac9k** [comment]
Nice find @pijyoi I would make this is a "good first issue" but this is sort of tricky to test the fix of.  

I guess the fix here is to replace `painter.drawText(text_pos[0], test_pos[1], self.text)` with `painter.drawText(QPointF(text_pos[0], text_pos[1], self.text)` ...wonder if it would be worthwhile to try and have `self.__project` return a `QPointF` instead (since it's building out the array to begin with, and we don't seem to use the 3rd element of it.

**Comment by pijyoi** [comment]
being addressed in #1889
